persistence-modules/spring-boot-persistence-2/pom.xml

```diff
@@ -43,7 +43,17 @@
     <dependencies>
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-jdbc</artifactId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>net.bytebuddy</groupId>
+            <artifactId>byte-buddy</artifactId>
```

Comment:
Is byte-buddy used?

Reply:
It's been used at runtime. Let me try to explain to you.
We are using hibernate-core (5.3.11.File). This dependency comes from spring-boot-starter-jpa (2.1.8.RELEASE). So, hibernate-core needs byte-buddy (1.9.16) for running correctly.
We are also using mockito-core (3.3.0), I'm not really sure where this dependency comes from. This mockito-core uses the 1.10.5 version from byte-buddy. So maven is omitting the 1.9.16 version (the one we need), and as the mockito-core is declared as test scope the byte-buddy.jar is not being included. This means that when you try to run the application you get a java.lang.ClassNotFoundException: net.bytebuddy.matcher.ElementMatchers
As I said, this only happens when you use the mvn spring-boot:run command.

If you want, I can remove this declaration, as I said mockito-core is declaring this dependency at test scope. So, all the tests are running fine